src/components/AddItem.js

```diff
@@ -39,42 +33,49 @@ const AddItem = () => {
       howSoon: e.target['how-soon'].value,
     };
 
-    getShoppingList(getToken())
+    shoppingLists()
+      .doc(getToken())
+      .get()
       .then((data) => {
-        if (data.docs.length) {
-          if (existingName(data.docs[0], newItem.name)) {
+        if (data.data()) {
+          if (existingName(data.data(), newItem.name)) {
             displayMessage(
               `The item: ${newItem.name} already exists!!`,
               'error',
             );
             return;
           }
-
-          shoppingLists()
-            .doc(data.docs[0].id)
-            .update({
-              items: firebase.firestore.FieldValue.arrayUnion(newItem),
-            });
-        } else {
-          shoppingLists().add({
-            token: getToken(),
-            items: [newItem],
-          });
         }
 
-        displayMessage('Successfully Added', 'success');
-        resetForm(setInputValue);
+        // add Item
+        shoppingLists()
+          .doc(getToken())
+          .set(
+            {
+              [uuid()]: newItem,
+            },
+            {
+              merge: true,
+            },
```

Comment:
Just to clarify, is `merge` used here to avoid overwriting objects in the document?

Reply:
Yeah true, it updates fields in the document or creates them if they don't exist (here we're using it to create)